go/cmd/dolt/cli/credentials.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+	"golang.org/x/term"
+)
+
+type UserPassword struct {
+	Username    string
+	Password    string
+	Unspecified bool // If true, the user and password were not provided by the user.
```

Comment:
Nit/personal pref: remove unnecessary negation here, it's confusing. Maybe change the field to "IsUserSupplied"?

Reply:
I changes the `Specifcied` because I think it's a good middle ground. It's an internal struct with a comment. Seems ok to me.